### ISSUE TITLE
Unrecoverable error on rizon download

### ISSUE THREAD

**Issue by itshoro** [namboy94/xdcc-dl#16]
Hi, when trying to use xdcc-dl on the default channel I've been experiencing an error message since yesterday. 
My setup hasn't changed, and I sadly have no idea on how to debug it properly.

Here is the verbose output

```sh
$ xdcc-dl "/msg CR-ARUTHA|NEW xdcc send #xxxx" -v

INFO: Download Limit set to: "unlimited"
INFO: Connecting to irc.rizon.net:6667
INFO: Timeout watcher started
WARNING: Unrecoverable Error: Is this IP banned?
INFO: Progress Printer started
INFO: Message sent without timeout
INFO: Progress Printer stopped
```

**Comment by GammaXA** [comment]
I have the same exact issue. The author needs to change how the script behaves in order to evade being labeled as "bot".

It seems like Rizon.net has a "bot detector" and I assume it blocks all xdcc-dl attempts no matter what device/IP address you try to use it from.

I think that the issue stems from xdcc-dl:
- Having a bot-like username. (aka a name pattern that is easy to detect)
(Solution: allow the user to pick a username as an option. Or have 2 lists of words and combine them randomly like BraveDino PickyPick...etc)
- When downloading multiple files xdcc-dl leaves and enters the same room again multiple times (according to another post here) and to ([https://abuse.rizon.net/akills/](https://abuse.rizon.net/akills/)). 
(Solution: the bot should stay and only leave after a random period from (1 min - 3 mins) to avoid bot detection. This would also allow the script to download the next files without having to leave the channel/server)

I wouldn't bet my cookies on the author replying back... Since the last release was in September 2019.

**Comment by namboy94** [comment]
This issue should be fixed in the latest release (5.0.0).

The username is now a random first and last name followed by two numbers. The username can also be set with the new ```--username``` command line parameter.

The program also adds random delays after connecting to a server and joining channels. The delay after connecting to the server can be set with the ```--channel-join-delay``` command line parameter.
